lib/templatesimplifier.cpp

```diff
@@ -3791,8 +3791,11 @@ void TemplateSimplifier::simplifyTemplates(
                 specializations,
                 maxtime,
                 expandedtemplates);
-            if (instantiated)
+            if (instantiated) {
                 mInstantiatedTemplates.push_back(*iter1);
+                mTemplateNamePos.clear(); // positions might be invalid after instantiations
```

Comment:
This was a cache to help improve the performance of the template simplifier. I wonder if we should only remove or adjust the template positions that comes after the instantiation.

Reply:
Probably. But I'm not familiar with this code at all. How much of a performance benefit was this cache?

Reply:
At the time, running with gtest headers, the time went from 48s to 5s. See here:

https://trac.cppcheck.net/ticket/9007#comment:12

Reply:
Then its probably fine.